client/extensions/woocommerce/woocommerce-services/state/service-settings/actions.js

```diff
@@ -1,18 +1,33 @@
 /**
  * Internal dependencies
  */
-export const SET_FORM_PROPERTY = 'SET_FORM_PROPERTY';
-export const SET_ALL_PRISTINE = 'SET_ALL_PRISTINE';
+import {
+	WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_UPDATE,
+	WOOCOMMERCE_SERVICES_SERVICE_SETTINGS_UPDATE_FIELD,
+} from '../action-types';
 
-export const setFormProperty = ( field, value ) => {
-	return {
-		type: SET_FORM_PROPERTY,
-		field,
-		value,
-	};
-};
+export const updateWcsShippingZoneMethod = (
+	siteId,
+	methodId,
+	methodType,
+	method,
+	successAction,
+	failureAction
+) => ( {
+	type: WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_UPDATE,
+	siteId,
+	methodId,
+	methodType,
+	method,
+	successAction,
+	failureAction,
+} );
 
-export const setAllPristine = ( pristineValue ) => ( {
-	type: SET_ALL_PRISTINE,
-	pristineValue,
+export const updateField = ( siteId, methodId, path, value ) => ( {
+	type: WOOCOMMERCE_SERVICES_SERVICE_SETTINGS_UPDATE_FIELD,
+	siteId,
+	methodId,
+	methodType: 'wc_services_usps', // Will work for the other methods too since they share the same reducer
```

Comment:
This action creator only works for USPS? Or does the comment mean it will work for others too?

Reply:
> This action creator only works for USPS? Or does the comment mean it will work for others too?

It works for the other WCS methods too.

> Looks like this is being used to determine if the shipping method is WCS. Would methodType: 'wc_services be enough here?

No. The `methodType` needs to match one of the `builtInShippingMethods`: https://github.com/Automattic/wp-calypso/blob/9fec0bd41ed09c6a53862c69ca292b26ccf96399/client/extensions/woocommerce/state/ui/shipping/zones/methods/reducer.js#L32
Since all the WCS methods use the same reducer, it doesn't matter what `methodType` is used, as long as it's in that object. I'll reword the comment to make it clearer.